README.md

```diff
@@ -1,3 +1,273 @@
+          --cluster=$GKE_CLUSTER \
+          --node-taints=node.config.status/stage=configuring:NoSchedule \
+          --zone=$GKE_ZONE # Or --region=$GKE_REGION
+        ```
+    * Verify the taint is applied to nodes in the pool:
+
+        ```bash
+        kubectl describe node <node-name-in-pool> | grep Taints
+        ```
+
+2.  **Run Init Container on Tainted Node:**
+    * Create a DaemonSet YAML (`tainted-config-daemonset.yaml`). This is similar to Part 1, but **adds the `tolerations` block**:
+
+        ```yaml
+        apiVersion: apps/v1
+        kind: DaemonSet
+        metadata:
+          name: tainted-config-daemonset
+          labels:
+            app: tainted-configurator
+        spec:
+          selector:
+            matchLabels:
+              app: tainted-configurator
+          template:
+            metadata:
+              labels:
+                app: tainted-configurator
+            spec:
+              # *** Add toleration to match the taint ***
+              tolerations:
+              - key: "node.config.status/stage"
+                operator: "Equal"
+                value: "configuring"
+                effect: "NoSchedule"
+              volumes:
+              - name: host-root-fs
+                hostPath:
+                  path: /
+                  type: Directory
+              # Allow access to host PID namespace if needed for specific tools
+              hostPID: true    
+              initContainers:
+              - name: apply-config-on-tainted
+                image: gcr.io/gke-release/debian-base # Small image with shell tools
+                securityContext:
+                  privileged: true # Still needs privilege for host changes
+                volumeMounts:
+                - name: host-root-fs
+                  mountPath: /host
+                command: ["/bin/sh", "-c"]
+                args:
+                - |
+                  echo "Applying configuration on tainted node..."
+                  # Example: Set a different sysctl value
+                  chroot /host sysctl -w vm.max_map_count=262144
+                  echo "Configuration applied."
+                  exit 0 # Ensure successful exit
+              containers:
+              - name: pause-container
+                image: us.gcr.io/gke-release/pause:latest
+        ```
+    * Apply this DaemonSet:
+
+        ```bash
+        kubectl apply -f tainted-config-daemonset.yaml
+        ```
+    * Verify pods run *only* on the tainted nodes:
+
+        ```bash
+        kubectl get pods -l app=tainted-configurator -o wide
+        ```
+    * Check logs to confirm configuration:
+
+        ```bash
+        POD_NAME=$(kubectl get pods -l app=tainted-configurator -o jsonpath='{.items[0].metadata.name}')
+        kubectl logs $POD_NAME -c apply-config-on-tainted
+        ```
+
+3.  **Remove the Taint:**
+    * Once configuration is complete, remove the taint from the node pool to make it generally schedulable.
```

Comment:
We actually want to always keep the taint on the node pool, just in case when the node pool scales up, new nodes always get the taint. Here we should use kubectl to remove taint from individual node. More easily, we can just do this in the daemonset after we verify the change has been made. Basically each node's daemonset pod untaint the node itself.